### PR TITLE
[RTM] Support clearing any proxy cache in system maintenance

### PR DESCRIPTION
**Pull request by Toflar**
Right now it only works for the `http_cache` because it purges the folder.
So I worked on the implementation in `FOSHttpCache` (https://github.com/FriendsOfSymfony/FOSHttpCache/pull/437) and the upcoming version 2.6 will now support clearing the reverse proxy cache completely.

This means whatever proxy you configured (by default it's the built in HttpCache so it still works just the same) it will be supported in the future 🎉

### CODE REVIEW

File: core-bundle/src/Resources/contao/config/config.php [hunk 1]
```diff
@@ -334,7 +329,11 @@
 		'symlinks' => array
 		(
 			'callback' => array('Automator', 'generateSymlinks')
-		)
+		),
+		'pages' => array
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/config/config.php]: Why did you move this down here?

Comment (Toflar) [reply core-bundle/src/Resources/contao/config/config.php]: Just to group the ones that output the number of files and the ones that do not :)

**Comment by leofeyer** [comment]
This one is RTM as soon as version 2.6 of https://github.com/FriendsOfSymfony/FOSHttpCache has been released.

**Comment by Toflar** [comment]
This should be possible now.

**Comment by leofeyer** [comment]
Version [2.6.0](https://github.com/FriendsOfSymfony/FOSHttpCache/releases/tag/2.6.0) has been released. 🎉

File: composer.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
         "doctrine/dbal": "^2.5",
         "doctrine/doctrine-bundle": "^1.7.2",
         "doctrine/doctrine-cache-bundle": "^1.3",
-        "friendsofsymfony/http-cache": "^2.5",
+        "friendsofsymfony/http-cache": "2.6.x-dev",
```
Comment (aschempp) [inline composer.json]: please update the requirement

File: core-bundle/src/Resources/contao/library/Contao/Automator.php [hunk 3]
```diff
@@ -144,18 +145,31 @@ public function purgeScriptCache()
...
-	 * @todo Replace this with a more sophisticated invalidation routine
 	 */
 	public function purgePageCache()
 	{
-		$strCacheDir = \StringUtil::stripRootDir(\System::getContainer()->getParameter('kernel.cache_dir'));
+		$container = \System::getContainer();
 
-		$objFolder = new \Folder($strCacheDir . '/http_cache');
-		$objFolder->purge();
+		if (!$container->has('fos_http_cache.cache_manager'))
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/library/Contao/Automator.php:154]: Shouldn't this case be handled before showing the option in the back end at all?

File: core-bundle/src/Resources/contao/library/Contao/Automator.php [hunk 4]
```diff
@@ -144,18 +145,31 @@ public function purgeScriptCache()
...
+			$this->log('Cannot purge the page cache (reverse proxy); invalid proxy configuration', __METHOD__, TL_ERROR);
+
+			return;
+		}
+
+		/** @var CacheManager $cacheManager */
+		$cacheManager = $container->get('fos_http_cache.cache_manager');
+
+		if (!$cacheManager->supports(CacheManager::CLEAR)) {
+			$this->log('Cannot purge the page cache (reverse proxy); invalid proxy configuration', __METHOD__, TL_ERROR);
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/library/Contao/Automator.php]: Same as above

**Review by aschempp — APPROVED** [review]

File: composer.json (hunk 2 above)
Comment (Toflar) [reply composer.json]: Done

File: core-bundle/src/Resources/contao/library/Contao/Automator.php (hunk 3 above)
Comment (Toflar) [reply core-bundle/src/Resources/contao/library/Contao/Automator.php:154]: I cannot. The maintenance module does not allow to have such pre-checks and I also don't think we should implement it because out-of-the-box it works. I only doesn't work if the configuration was adjusted and then I think it's better to log something than to just hide the option all together.

**Review by Toflar — COMMENTED** [review]
@leofeyer this is ready then.

**Comment by leofeyer** [comment]
Thank you @Toflar.
